### PR TITLE
Issue #SB-16613 feat: Issue #SB-16613 feat: perfLog implementation

### PR DESCRIPTION
**Pull request by anoophm**

### CODE REVIEW

File: services/perfLogger/perfLogger.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import { SystemQueue, ISystemQueue } from './../../services/queue';
+
+const INITIAL_TRIGGER = 15 * 60 * 1000; // trigger first job after 15 min  
+const DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000; // trigger jobs every 24 hours after first trigger
+const MONTH_IN_MILLISECONDS = 30 * 24 * 60 * 60 * 1000; // used in archive job to remove logs which are older than last 30 days
+const LOG_QUERY_LIMIT = 1000;
+const DB_NAME = 'perf_log';
+const LAST_PERF_LOG_PROCEEDED_ON = 'LAST_PERF_LOG_PROCEEDED_ON';
+const REQUIRED_SYSTEM_QUEUE_TASK = ["IMPORT", "DOWNLOAD"]; // should be moved to openrap-sunbirded-plugin to make it customizable
+const DEFAULT_LAST_SYNC_TIME = 1585282913052;
```
Comment (HarishGangula) [inline services/perfLogger/perfLogger.ts]: What is this time ?

File: services/perfLogger/perfLogger.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import { EventManager } from "@project-sunbird/ext-framework-server/managers/EventManager";
+import { SystemQueue, ISystemQueue } from './../../services/queue';
+
+const INITIAL_TRIGGER = 15 * 60 * 1000; // trigger first job after 15 min  
+const DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000; // trigger jobs every 24 hours after first trigger
+const MONTH_IN_MILLISECONDS = 30 * 24 * 60 * 60 * 1000; // used in archive job to remove logs which are older than last 30 days
+const LOG_QUERY_LIMIT = 1000;
+const DB_NAME = 'perf_log';
+const LAST_PERF_LOG_PROCEEDED_ON = 'LAST_PERF_LOG_PROCEEDED_ON';
+const REQUIRED_SYSTEM_QUEUE_TASK = ["IMPORT", "DOWNLOAD"]; // should be moved to openrap-sunbirded-plugin to make it customizable
```
Comment (HarishGangula) [inline services/perfLogger/perfLogger.ts]: we need to add delete  also

File: services/perfLogger/perfLogger.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        }
+        if(limit){
+            query['limit'] = limit;
+        }
+        if (fields) {
+            query['fields'] = fields;
+        }
+        return this.dbSDK.find(DB_NAME, query).then(data => data.docs);
+    }
+    private getStartAndEndEpochTime(dateInEpoch = Date.now()): {startTime: number, endTime: number}{
```
Comment (HarishGangula) [inline services/perfLogger/perfLogger.ts]: This is used for ?

File: services/perfLogger/perfLogger.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+export class PerfLogger {
+    @Inject private dbSDK: DataBaseSDK;
+    @Inject private settingSDK: SettingSDK;
+    constructor() {
+        timer(INITIAL_TRIGGER, DAY_IN_MILLISECONDS).subscribe(this.handleTimerEvent.bind(this))
+        EventManager.subscribe(SystemQueue.taskCompleteEvent, (data: ISystemQueue) => { // this should be moved to openrap-sunbirded-plugin
+            if(!_.includes(REQUIRED_SYSTEM_QUEUE_TASK, data.type)){
+                return;
+            }
+            // TODO: divide time by contentSize
```
Comment (HarishGangula) [inline services/perfLogger/perfLogger.ts]: This may not be on time by ContentSize may include others also

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: services/perfLogger/perfLogger.ts (hunk 2 above)
Comment (anoophm) [reply services/perfLogger/perfLogger.ts]: in openrap-sunbird-plugin

File: services/perfLogger/perfLogger.ts (hunk 1 above)
Comment (anoophm) [reply services/perfLogger/perfLogger.ts]: to take as default, after first process last processed time will be stored in setting sdk

File: services/perfLogger/perfLogger.ts (hunk 4 above)
Comment (anoophm) [reply services/perfLogger/perfLogger.ts]: will move this to openrap-sunbirded-plugin. we can add custom logic for each type

File: services/perfLogger/perfLogger.ts (hunk 3 above)
Comment (anoophm) [reply services/perfLogger/perfLogger.ts]: used for querying from database and aggregating log for each day

**Pull request closed without merging**